src/mast/ast.rs

```diff
@@ -5,14 +5,10 @@ use super::MastCtx;
 impl Expr {
     /// Convert an expression to another expression, with the same span and a regenerated node id.
     pub fn to_mast(&self, ctx: &mut MastCtx, kind: &ExprKind) -> Expr {
-        let node_id = ctx.next_node_id();
-        let span = self.span;
-        let kind = kind.clone();
-
         Expr {
-            node_id,
-            span,
-            kind,
+            node_id: ctx.next_node_id(),
+            kind: kind.clone(),
```

Comment:
It will also replace:
- BigInt (if it is a generic parameter)
- FnCall (the generic arguments will be replaced)

So anything that will need to be converted to concrete type will be replaced.